reactioncommerce/reaction
# [WIP] feat: use Segment.io analytics from starterkit in the main app

Note: currently blocked on https://github.com/reactioncommerce/reaction/pull/4584

Resolves #4611 
Impact: **minor**  
Type: **feature**

## Issue
With the removal of our analytics plugins from core, there is currently no way to track various analytics.

## Solution
This PR ports the Segment.io/Google Analytics tracking logic from Starterkit. `trackProductListViewed` and `trackProductClicked` are fully functional. `trackProductViewed` will need to be added in later, once the PDP has been fully ported to GraphQL.

## Breaking changes
None


## Testing
1. Run the app as usual, without adding a Meteor settings.json file. Visit home page, tag page and click a product. Confirm no client errors related to segment tracking (no regressions).
2. Get a Segment.io write key and run the app with a settings.json:
```
{
  "public": {
    "segmentAnalyticsWriteKey": "SEGMENT_KEY_HERE"
  }
}
```
`METEOR_DISABLE_OPTIMISTIC_CACHING=1 REACTION_LOG_LEVEL="DEBUG" reaction --settings private/settings/settings.json`
3. Segment.io -> Sources -> Debugger
4. Visit homepage and confirm page view event for "/" and "Product List Viewed" event in Segment. Confirm correct product data in event.
5. Switch languages and confirm correct currency and pricing data is sent to Segment with the "Product List Viewed" event
5. Visit tag page, confirm page view for "/tag/..." and "Product List Viewed" event with correct data
6. Click a product in the grid and confirm "Product Clicked" event with correct data
7. Click various pages and confirm page views in segment
8. Open search confirm that when you stop typing, if there are results, a "Product List Viewed" event is sent to Segment with the right products & data
9. Click a product in the search grid and confirm "Product Clicked" event with correct data
